test/suite/test_compact11.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wttest
+from wiredtiger import stat
+
+kilobyte = 1024
+
+# test_compact11.py
+# This test creates:
+#
+# - One table with the first 1/4 of keys deleted and obsolete and the last 10th of the keys
+#   removed with fast truncate but not yet obsolete.
+#
+# It checks that:
+#
+# - We correctly rewrite pages in WT_REF_DELETED state but are still on disk.
+class test_compact11(wttest.WiredTigerTestCase):
+    create_params = 'key_format=i,value_format=S,allocation_size=4KB,leaf_page_max=32KB,leaf_value_max=16MB'
+    conn_config = 'cache_size=10MB,statistics=(all),debug_mode=(background_compact),verbose=[compact:4]'
+    uri_prefix = 'table:test_compact11'
+
+    table_numkv = 10 * 1000
+    value_size = kilobyte # The value should be small enough so that we don't create overflow pages.
+    
+    def get_size(self, uri):
+        stat_cursor = self.session.open_cursor('statistics:' + uri, None, 'statistics=(all)')
+        size = stat_cursor[stat.dsrc.block_size][2]
+        stat_cursor.close()
+        return size
+
+    def truncate(self, uri, key1, key2):
+        lo_cursor = self.session.open_cursor(uri)
+        hi_cursor = self.session.open_cursor(uri)
+        lo_cursor.set_key(key1)
+        hi_cursor.set_key(key2)
+        try:
+            err = self.session.truncate(None, lo_cursor, hi_cursor, None)
+        except WiredTigerError as e:
+            if wiredtiger_strerror(WT_ROLLBACK) in str(e):
```

Comment:
Since we are not doing anything with those errors, we could simple raise the exception all the time.
After thinking about it again, we can simply expect a valid truncate at all times.

Reply:
Fair enough, I was following the truncate python tests as examples.

Reply:
I changed it to just rely on https://github.com/wiredtiger/wiredtiger/blob/714fca3699e048e5dbed945d81c1d1b004f3ad7e/test/suite/test_compact12.py#L62-L63 failing if there is an error.